spec/datadog/core/telemetry/heartbeat_spec.rb

```diff
@@ -16,11 +16,11 @@
 
   describe '.new' do
     context 'when using default settings' do
-      subject(:heartbeat) { described_class.new(&block) }
+      subject(:heartbeat) { described_class.new(interval: interval, &block) }
```

Comment:
Minor: I suggest propagating the `_seconds` prefix internally as well:

```suggestion
      subject(:heartbeat) { described_class.new(heartbeat_interval_seconds: heartbeat_interval_seconds, &block) }
```